Add various explanations and translations (Fixes #56)

florence opened the pull request:
This commit attempt to build some alternatives to `lift-above-baseline`
and `drop-below-ascent`. To do this introduces a `translate` function, which
shifts pict's around. In addition it introduces the function `explain`,
which attempts to annotate picts to show their baseline and ascent.

I'm not 100% sure these functions are what we want though: I couldn't divine the intended behavior
of `lift-above-baseline`, so this is a shot at something I think is reasonable and possibly useful?

The new doc's look something like this:

![Screen Shot 2019-12-13 at 4 29 29 PM](https://user-images.githubusercontent.com/1413610/70836488-e53d8b00-1dc5-11ea-99c8-9ef790f42742.png)
![Screen Shot 2019-12-13 at 4 29 35 PM](https://user-images.githubusercontent.com/1413610/70836489-e53d8b00-1dc5-11ea-9c55-3de39db60901.png)
![Screen Shot 2019-12-13 at 4 29 57 PM](https://user-images.githubusercontent.com/1413610/70836490-e53d8b00-1dc5-11ea-8ac7-51842efeb90a.png)
![Screen Shot 2019-12-13 at 4 30 11 PM](https://user-images.githubusercontent.com/1413610/70836491-e53d8b00-1dc5-11ea-84a1-a4733969fa68.png)

rfindler commented:
I think explain needs to make the bounding box bigger. At least as big as
any drawing would do, perhaps with some padding.

florence commented:
Okay I did that and also added an `explain-child`. `explain-child` seems to be mishbehaving under rotations and shears, but its only off by a constant shift. The transformation matricies I get out look correct, so I haven't the foggiest clue why its odd. But at the moment it's good enough for government work, so I'm inclined to look into in more once someone complains.

The transformation stuff I added will hopefully be useful for future functions (like a variant of `pin-over` that respects scaling). 

In while doing this I noticed that the `sxy` and `syx` arguments for `make-child` were flipped in the docs, so I fixed that too.

The new docs look like:
![Screen Shot 2019-12-16 at 2 49 33 PM](https://user-images.githubusercontent.com/1413610/70942306-01366c00-2014-11ea-8a65-8dc963026164.png)


The new margins made the explanations of the clipping functions a little odd:

![Screen Shot 2019-12-16 at 2 55 49 PM](https://user-images.githubusercontent.com/1413610/70942397-35119180-2014-11ea-99cd-51a88acdc4d6.png)

But that might be okay?

Also FWIW this is what the "slightly off" looks like for rotations:

![Screen Shot 2019-12-16 at 2 57 22 PM](https://user-images.githubusercontent.com/1413610/70942492-65f1c680-2014-11ea-9e2f-5e6a17c976fb.png)

(where its less "off" for rotations in-between `0` and `pi`)

mflatt commented:
This all seems good to me.

I'm interested to look into `explain-child` and rotations after its merged, but I don't know if I would actually get around to it soon.
